### ISSUE TITLE
Deprecated `moduleResolution: "node"` in published src/tsconfig.json triggers TS 5.9 deprecation warning

### ISSUE THREAD

**Issue by netoxp70** [RetellAI/retell-typescript-sdk#505]
Starting in TypeScript 5.9, `moduleResolution: "node"` (alias for `node10`) is
deprecated and scheduled for removal in TypeScript 7.0. The `src/tsconfig.json`
shipped in the published package still uses it:

https://github.com/RetellAI/retell-typescript-sdk/blob/main/src/tsconfig.json

```json
{
  "include": ["index.ts"],
  "compilerOptions": {
    "target": "ES2015",
    "lib": ["DOM", "DOM.Iterable", "ES2018"],
    "moduleResolution": "node"
  }
}

**Comment by fedorkar** [comment]
Thanks for reporting this. Fixed in [retell-sdk 5.66.1](https://github.com/RetellAI/retell-typescript-sdk/releases/tag/v5.66.1), now available on npm.

The published source config now uses `module: "Node16"` and `moduleResolution: "Node16"`. This removes the deprecation while preserving TypeScript 4.9 support. We verified the published tarball with TypeScript 4.9.5, 5.9.3, and 6.0.2, including CommonJS and ESM consumer projects.

One clarification: this deprecation is reported by TypeScript 6.0; we did not reproduce it with 5.9.3. The affected config is used when navigating into the SDK's bundled source. Normal SDK imports do not inherit it, and the SDK's runtime build configuration is unchanged.

Please upgrade to 5.66.1 or later. Closing this as fixed; if you still encounter this diagnostic after upgrading, please share your exact compiler version and diagnostic.
